lib/charms/alertmanager_k8s/v0/alertmanager_dispatch.py

```diff
@@ -70,10 +69,21 @@ class RelationManagerBase(Object):
         name (str): consumer's relation name
     """
 
-    def __init__(self, charm: CharmBase, relation_name):
+    def __init__(self, charm: CharmBase, relation_name: str, relation_role: RelationRole):
```

Comment:
Not checking the interface? -1

Reply:
My impression was that hardcoding the interface here is still debatable :)

Reply:
Your impression is likely wrong ;-)
Yesterday I showed how you cannot rely on the interface in `metadata.yaml`, remember?

Reply:
Yes. I am not convinced yet.
Although if we already hardcode the relation name there's not much sense to keep the interface name open.

Reply:
I think you are mixing things up a bunch: the interface name is fixed because otherwise `juju relate` simply does not work. The relation name is something that, as discussed in the last plenary, makes sense fixed for things like observability relations, and not for things like database relations.

Reply:
Added.